homeassistant/components/microbees/climate.py

```diff
@@ -0,0 +1,142 @@
+"""Climate integration microBees."""
+from typing import Any
+
+from microBeesPy import Sensor
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import MicroBeesUpdateCoordinator
+from .entity import MicroBeesActuatorEntity
+
+CLIMATE_PRODUCT_IDS = {76, 78}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the microBees climate platform."""
+    coordinator: MicroBeesUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ].coordinator
+    async_add_entities(
+        MBClimate(
+            coordinator,
+            bee_id,
+            bee.actuators[0].id,
+            next(
+                filter(
+                    lambda x: x.deviceID == (762 if bee.productID == 76 else 782),
+                    bee.sensors,
+                )
+            ).id,
```

Comment:
This is difficult to unpack what is going on here.  I'd make it a named function without magic numbers that better shows the intent.

Alternatively it might make sense to not use a genexpr here and build a list of entities to pass to `async_add_entities`

Reply:
It's just a filter to find the correct sensor based on deviceId. we can use a function but it will not be much different from the actual solution